.github/workflows/ci-cd.yml

```
@@ -517,3 +529,6 @@ jobs:
       uses: re-actors/alls-green@05ac9388f0aebcb5727afa17fcccfecd6f8ec5fe  # v1.2.2
       with:
         jobs: ${{ toJSON(needs) }}
+        # Needed to not fail on skipped 'push' events for the branches in
+        # upstream repository as they already covered by 'pull_request' event
+        allowed-skips: lint, tests
```

This is what I did in CPython. It is more granular this way.
```suggestion
        allowed-skips: >-
          ${{
            (
              github.repository_id != needs.pre-setup.outputs.upstream-repository-id
              || github.event_name != 'push'
              || github.ref_name == github.event.repository.default_branch
            )
            && 'lint, tests'
            || ''
          }}
```